python/cudf/cudf/dataframe/dataframe.py

```diff
@@ -1062,8 +1062,23 @@ def take(self, positions, ignore_index=False):
                 out.index = RangeIndex(len(out))
             else:
                 out.index = self.index.take(positions)
+
+        if isinstance(self.index, cudf.MultiIndex):
+            out._index = self.index.take(positions)
+
         return out
 
+    def take_columns(self, positions):
+        positions = Series(positions)
+        column_names = list(self._cols.keys())
+        column_values = list(self._cols.values())
+        result = DataFrame()
+        for idx in range(len(positions)):
+            result[column_names[positions[idx]]] = column_values[
+                positions[idx]]
+        result._index = self._index
+        return result
```

Comment:
What's the difference between this function and just using `take` with `ignore_index=True`?

Reply:
This is a columnar-gather function, perhaps the name is not descriptive enough. `positions` should be a list-like of integer values, which will then return a new DataFrame composed of the columns specified by the integral order of `positions`.

```
In [4]: cudf.DataFrame({'x': [1], 'y': [2], 'z': [3]}).take_columns([0, 2]).to_p
   ...: andas()
Out[4]:
   x  z
0  1  3
```